### PR TITLE
feat: add query param to disable the nft sections of the app

### PR DESCRIPTION
**Pull request by JFrankfurt**
This PR adds a query parameter `disableNFTs=<boolean>` that can be used to hide the NFT-related parts of the web app from the user. It is intended to be used by dapp browser webviews which are not allowed to surface NFTs in app stores. This flag is saved to session storage with a jotai storage atom. (see `src/state/application/atoms.ts`)

demo: https://interface-git-fork-jfrankfurt-nft-tab-disabled-query-uniswap.vercel.app/#/?disableNFTs=true

JIRA: https://uniswaplabs.atlassian.net/jira/software/c/projects/WEB/boards/9?modal=detail&selectedIssue=WEB-3056&assignee=604fcc8b81b82500682e18e8

### CODE REVIEW

File: src/pages/App.tsx [hunk 1]
```diff
@@ -146,6 +149,10 @@ export default function App() {
     setScrolledState(false)
   }, [pathname])
 
+  useEffect(() => {
+    setShouldDisableNFTRoutes(Boolean(search && search.includes('disableNfts=true')))
+  }, [search, setShouldDisableNFTRoutes])
```
Comment (grabbou) [inline src/pages/App.tsx:154]: Why do we need an atom here? Can't we simply use `useLocation` in a NavBar component and move this logic there?

**Review by grabbou — CHANGES_REQUESTED** [review]

File: src/pages/App.tsx (hunk 1 above)
Comment (JFrankfurt) [reply src/pages/App.tsx]: the effect could be run anywhere, since it's just consuming react-router-dom

The atom is used because the `shouldDisableNFTRoutes` state is consumed at the NavBar to disable user interaction and also at the nft-related `Route` components below to disable further direct nav.

**Review by JackShort — APPROVED** [review]

**Comment by tinaszheng** [comment]
image
you may want to hide this button too?

**Comment by tinaszheng** [comment]
also another nit, if we're disabling NFTs, users are still going to see this intro screen for the first time - seems weird to disable NFTs but still have the copy that says "Trade crypto & NFTs". maybe disable the intro screen too 

image

File: src/pages/App.tsx [hunk 2]
```diff
@@ -146,6 +149,11 @@ export default function App() {
     setScrolledState(false)
   }, [pathname])
 
+  const [searchParams] = useSearchParams()
+  useEffect(() => {
+    if (searchParams.has('disableNFTs')) setShouldDisableNFTRoutes(true)
```
Comment (tinaszheng) [inline src/pages/App.tsx:154]: you should probably do an explicit comparison here to = true since if users pass in `disableNFTs=false` this would be wrong

**Review by grabbou — CHANGES_REQUESTED** [review]
Thanks for updates, just left another comment (I think I am still concerned about the same part as before, but requesting changes once again to make it explicit Github UI-wise).

File: src/pages/App.tsx [hunk 3]
```diff
@@ -146,6 +149,15 @@ export default function App() {
     setScrolledState(false)
   }, [pathname])
 
+  const [searchParams] = useSearchParams()
+  useEffect(() => {
+    if (searchParams.get('disableNFTs') === 'true') {
+      setShouldDisableNFTRoutes(true)
```
[the remark this answers is not in the document]
Comment (JFrankfurt) [reply src/pages/App.tsx:155]: Deriving a state flag from the parameter and using the sessionstorage atom allows us to persist the disabled experience without ensuring that all navigation through the app persists query parameters. So if someone goes to the LP page and creates a new position that has certain min/max tick values and a fee encoded in the query string, and that code doesn't explicitly ensure persistence of existing parameters, then the NFT experience exclusion will cease. 

The ask here is that we create a query parameter that can be linked from a wallet browser that will disable the NFT experience. The most reliable way to do that is to persist it to the user's session instead of auditing existing routing code and then remembering that we need to ensure param persistence in future routing code.

[the remark this answers is not in the document]
Comment (JFrankfurt) [reply src/pages/App.tsx:155]: yes exactly

I'll add a doc comment to the atom to explain what's going on.

[the remark this answers is not in the document]
Comment (grabbou) [reply src/pages/App.tsx:155]: Would it make sense to - rather than creating a new atom - extend our existing `featureFlag` mechanism?

TL;DR add a new feature flag for NFT section enabled / disabled and provide a mechanism for setting feature flag values by passing them as a query string - ideally, as a generic solution - but can be made for this one specifically too.

I would do a combination of this hook that checks for query parameters and combine it with `useUpdateFlag`. If generic, I would look into URL encoding a JSON as as `flags` query param. If dedicated, I'd stick to `disableNFTs`.

**Review by tinaszheng — CHANGES_REQUESTED** [review]

File: src/pages/Landing/index.tsx [hunk 4]
```diff
@@ -342,7 +346,9 @@ export default function Landing() {
             <Glow />
           </GlowContainer>
           <ContentContainer isDarkMode={isDarkMode}>
-            <TitleText isDarkMode={isDarkMode}>Trade crypto & NFTs with confidence</TitleText>
+            <TitleText isDarkMode={isDarkMode}>
```
[the remark this answers is not in the document]
Comment (tinaszheng) [reply src/pages/Landing/index.tsx:349]: wait i think it might be because your logic is wrong here - shouldn't it be `!shouldDisableNFTRoutes ? '& NFTs ' : ''`

[the remark this answers is not in the document]
Comment (JFrankfurt) [reply src/pages/Landing/index.tsx:349]: lul good catch, thanks

[the remark this answers is not in the document]
Comment (JFrankfurt) [reply src/pages/Landing/index.tsx:349]: image

**Review by tinaszheng — APPROVED** [review]
LFG 🔥 

can you update your PR description with implementation details for the `atom` and session storage

File: src/state/application/atoms.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { atomWithStorage, createJSONStorage } from 'jotai/utils'
+
+/*
+  note: 
+  We should consider a generic sessionStorage abstraction if this pattern becomes common. (i.e., for future promo dismissals like the tax service discounts or FoR launch notification)
```
Comment (grabbou) [inline src/state/application/atoms.ts:5]: nit: please make sure this is uppercase where necessary (new sentence or heading) and has periods at the endow each sentence.

There are also a few minor issues, such as period after "common" and "FoR" (prefer explicit name).

**Review by grabbou — APPROVED** [review]
Looking good, thanks for the PR! Left last nit, but it's good to go otherwise!
